applications/app/controllers/CrosswordsController.scala

```diff
@@ -350,7 +354,25 @@ class CrosswordEditionsController(
     "crosswords/series/quiptic",
   ).mkString("|")
 
-  private def parseCrosswords(response: SearchResponse): EditionsCrosswordRenderingDataModel =
-    EditionsCrosswordRenderingDataModel(response.results.flatMap(_.crossword))
+  private def parseCrosswords(response: SearchResponse): Seq[EditionsCrosswordRenderingDataModel] = {
+    response.results.flatMap(_.crossword).map { crossword =>
+      EditionsCrosswordRenderingDataModel(
+        id = s"crosswords/${crossword.`type`.name}/${crossword.number}",
+        number = crossword.number,
+        name = crossword.name,
+        creator = crossword.creator,
+        date = crossword.date.toJoda.getMillis,
+        webPublicationDate = crossword.date.toJoda.getMillis,
+        entries =
+          crossword.entries.map(entry => EditionsCrosswordEntry.fromCrosswordEntry(entry, crossword.solutionAvailable)),
```

Comment:
I think we can make this immutable here since we're not mutating anything?

```suggestion
          crossword.entries.toSeq.map(entry => EditionsCrosswordEntry.fromCrosswordEntry(entry, crossword.solutionAvailable)),
```